src/actions/account.js

```diff
@@ -0,0 +1,88 @@
+// @flow
+import type {Dispatch} from 'redux'
+
+import walletManager from '../crypto/wallet'
+import type {State} from '../state'
+
+// start fetching account balance
+const _startFetching = () => ({
+  type: 'START_FETCHING_ACCOUNT_STATE',
+  path: ['accountState', 'isFetching'],
+  payload: null,
+  reducer: (state, isFetching) => true,
+})
+
+const _endFetching = () => ({
+  type: 'END_FETCHING_ACCOUNT_STATE',
+  path: ['accountState', 'isFetching'],
+  payload: null,
+  reducer: (state, isFetching) => false,
+})
+
+const _setAccountValue = (value) => ({
+  type: 'SET_ACCOUNT_VALUE',
+  path: ['accountState', 'value'],
+  payload: value,
+  reducer: (state, value) => value,
+})
+
+const _setAccountPools = (pools) => ({
+  type: 'SET_ACCOUNT_POOLS',
+  path: ['accountState', 'delegation', 'pools'],
+  payload: pools,
+  reducer: (state, pools) => pools,
+})
+
+const _clearAccountState = () => ({
+  type: 'CLEAR_ACCOUNT_STATE',
+  path: ['accountState'],
+  payload: null,
+  reducer: (state) => {
+    return {
+      isFetching: false,
+      lastFetchingError: null,
+      delegation: {pools: []},
+      value: 0,
+      counter: 0,
+      last_rewards: {
+        epoch: 0,
+        reward: 0,
+      },
+    }
+  },
+})
+
+const _setLastError = (error) => ({
+  type: 'SET_LAST_FETCHING_ERROR',
+  path: ['accountState', 'lastFetchingError'],
+  payload: error,
+  reducer: (state, error) => error,
+})
+
+export const fetchAccountState = () => async (
+  dispatch: Dispatch<any>,
+  getState: () => State,
+) => {
+  if (getState().accountState.isFetching) {
+    return
+  }
+  dispatch(_clearAccountState())
+  dispatch(_startFetching())
+  try {
+    // $FlowFixMe (v-almonacid): not sure how to correctly handle union types
+    const accountStateResp = await walletManager.fetchAccountState()
+    const accountState = Object.keys(accountStateResp).map(
+      (key) => accountStateResp[key],
+    )[0]
+    if ('error' in accountState) throw new Error(accountState.error)
```

Comment:
This fixes the Flow issue so you can remove the `// $FlowFixMe`
```suggestion
    if (accountState.error != null) throw new Error(accountState.error)
```